jetty-server/src/main/java/org/eclipse/jetty/server/LowResourceMonitor.java

```diff
@@ -398,4 +423,117 @@ private String low(String reasons, String newReason)
     private static class LRMScheduler extends ScheduledExecutorScheduler
     {
     }
+
+    interface LowResourceCheck
+    {
+        boolean isLowOnResources();
+    }
+
+    //------------------------------------------------------
+    // default implementations for backward compat
+    //------------------------------------------------------
+
+    public class MainThreadPoolLowResourceCheck implements LowResourceCheck
+    {
+        private String reason;
+
+        public MainThreadPoolLowResourceCheck()
+        {
+            // no op
+        }
+
+        @Override
+        public boolean isLowOnResources()
+        {
+            ThreadPool serverThreads = _server.getThreadPool();
+            if (serverThreads.isLowOnThreads())
+            {
+                reason="Server low on threads: "+serverThreads;
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public String toString()
+        {
+            return reason;
+        }
+    }
+
+    public class ConnectorsThreadPoolLowResourceCheck implements LowResourceCheck
+    {
+        private String reason;
+
+
+        public ConnectorsThreadPoolLowResourceCheck()
+        {
+            // no op
+        }
+
+        @Override
+        public boolean isLowOnResources()
+        {
+            ThreadPool serverThreads = _server.getThreadPool();
+
+            int connections=0;
+
+            for(Connector connector : getMonitoredConnectors())
+            {
+                connections+=connector.getConnectedEndPoints().size();
+                Executor executor = connector.getExecutor();
+                if (executor instanceof ThreadPool && executor!=serverThreads)
+                {
+                    ThreadPool connectorThreads=(ThreadPool)executor;
+                    if (_monitorThreads && connectorThreads.isLowOnThreads())
+                    {
+                        reason ="Connector low on threads: "+connectorThreads;
+                        return true;
+                    }
+                }
+            }
+            if (_maxConnections>0 && connections>_maxConnections)
+            {
+                reason ="Max Connections exceeded: "+connections+">"+_maxConnections;
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public String toString()
+        {
+            return reason;
```

Comment:
hmmm I  know I suggested doing it like this.... but now looking at it, it is rather ugly.

So I think that `toString` should describe the `LowResourceCheck` and we need a better way to get the reason.
Perhaps just a `getReason` method? or return the reason string from isLowResources?

Reply:
:) yup definitely I would prefer ```getReason```